x-pack/test_serverless/api_integration/services/svl_cases/api.ts

```diff
@@ -0,0 +1,233 @@
+import { ConnectorTypes } from '@kbn/cases-plugin/common/types/domain';
+import { CasesFindResponse } from '@kbn/cases-plugin/common/types/api';
+import { kbnTestConfig, kibanaTestSuperuserServerless } from '@kbn/test';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function SvlCasesApiServiceProvider({ getService }: FtrProviderContext) {
+  const kbnServer = getService('kibanaServer');
+  const supertest = getService('supertest');
+
+  interface User {
+    username: string;
+    password: string;
+    description?: string;
+    roles: string[];
+  }
+
+  const superUser: User = {
+    username: 'superuser',
+    password: 'superuser',
+    roles: ['superuser'],
+  };
+
+  const defaultUser = {
+    email: null,
+    full_name: null,
+    username: kbnTestConfig.getUrlParts(kibanaTestSuperuserServerless).username,
+  };
+
+  /**
+   * A null filled user will occur when the security plugin is disabled
+   */
+  const nullUser = { email: null, full_name: null, username: null };
+
+  const findCommon = {
+    page: 1,
+    per_page: 20,
+    total: 0,
+    count_open_cases: 0,
+    count_closed_cases: 0,
+    count_in_progress_cases: 0,
+  };
+
+  const findCasesResp: CasesFindResponse = {
+    ...findCommon,
+    cases: [],
+  };
+
+  return {
+    setupAuth({
+      apiCall,
+      headers,
+      auth,
+    }: {
+      apiCall: SuperTest.Test;
+      headers: Record<string, unknown>;
+      auth?: { user: User; space: string | null } | null;
+    }): SuperTest.Test {
+      if (!Object.hasOwn(headers, 'Cookie') && auth != null) {
+        return apiCall.auth(auth.user.username, auth.user.password);
+      }
+
+      return apiCall;
+    },
+
+    getSpaceUrlPrefix(spaceId: string | undefined | null) {
+      return spaceId && spaceId !== 'default' ? `/s/${spaceId}` : ``;
+    },
+
+    async deleteAllCaseItems() {
+      await Promise.all([
+        this.deleteCasesByESQuery(),
+        this.deleteCasesUserActions(),
+        this.deleteComments(),
+        this.deleteConfiguration(),
+        this.deleteMappings(),
+      ]);
+    },
+
+    async deleteCasesUserActions(): Promise<void> {
+      await kbnServer.savedObjects.clean({ types: ['cases-user-actions'] });
+    },
+
+    async deleteCasesByESQuery(): Promise<void> {
+      await kbnServer.savedObjects.clean({ types: ['cases'] });
+    },
+
+    async deleteComments(): Promise<void> {
+      await kbnServer.savedObjects.clean({ types: ['cases-comments'] });
+    },
+
+    async deleteConfiguration(): Promise<void> {
+      await kbnServer.savedObjects.clean({ types: ['cases-configure'] });
+    },
+
+    async deleteMappings(): Promise<void> {
+      await kbnServer.savedObjects.clean({ types: ['cases-connector-mappings'] });
+    },
+
+    /**
+     * Return a request for creating a case.
+     */
+    getPostCaseRequest(owner: string, req?: Partial<CasePostRequest>): CasePostRequest {
```

Comment:
I think I created these helper functions when I was not so familiar with serverless tests, and created helper functions to use cases APIs. 

But can we use `cases` service for apis  `createCase, postCase, getCase and findCases` ?

Reply:
> But can we use cases service for apis createCase, postCase, getCase and findCases ?

If they are the same / work the same on serverless, then yes. If there are adjustments needed to make these service methods work in serverless, then the serverless service is the right way to go (avoid `ifServerless doA else doB`).

Reply:
Thanks Robert. 
There are not major changes except serverless needs `x-elastic-internal-origin` request header. I will try to clean up this helpers and use `cases` api where possible in a separate PR.

Reply:
I agree with @js-jankisalvi that if possible is better to use the `cases` service.